plugin/plugin_test.go

```diff
@@ -0,0 +1,80 @@
+package junoplugin_test
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/NethermindEth/juno/blockchain"
+	"github.com/NethermindEth/juno/clients/feeder"
+	"github.com/NethermindEth/juno/db/pebble"
+	"github.com/NethermindEth/juno/mocks"
+	junoplugin "github.com/NethermindEth/juno/plugin"
+	adaptfeeder "github.com/NethermindEth/juno/starknetdata/feeder"
+	"github.com/NethermindEth/juno/sync"
+	"github.com/NethermindEth/juno/utils"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/mock/gomock"
+)
+
+const timeout = time.Second
+
+func TestPlugin(t *testing.T) {
+	mockCtrl := gomock.NewController(t)
+	t.Cleanup(mockCtrl.Finish)
+
+	plugin := mocks.NewMockJunoPlugin(mockCtrl)
+
+	mainClient := feeder.NewTestClient(t, &utils.Mainnet)
+	mainGw := adaptfeeder.New(mainClient)
+
+	integClient := feeder.NewTestClient(t, &utils.Integration)
+	integGw := adaptfeeder.New(integClient)
+
+	testDB := pebble.NewMemTest(t)
+
+	// sync to integration for 2 blocks
+	for i := range 2 {
+		su, block, err := integGw.StateUpdateWithBlock(context.Background(), uint64(i))
+		require.NoError(t, err)
+		plugin.EXPECT().NewBlock(block, su, gomock.Any())
+	}
+	bc := blockchain.New(testDB, &utils.Integration)
+	synchronizer := sync.New(bc, integGw, utils.NewNopZapLogger(), 0, false).WithPlugin(plugin)
+
+	ctx, cancel := context.WithTimeout(context.Background(), timeout)
+	require.NoError(t, synchronizer.Run(ctx))
+	cancel()
+
+	t.Run("resync to mainnet with the same db", func(t *testing.T) {
```

Comment:
Can you please elaborate on this? I'm don't understand why we need it

Reply:
Sure, I just wanted to add a test where we sync a couple of blocks, then revert some, then resume syncing again. I seen that this was implemented in one of the reorg tests so just re-used some test code from there. The specific network we are using doesn't matter, just that the pluging functions are called as expected.